adabot/lib/circuitpython_library_validators.py

```diff
@@ -439,9 +439,11 @@ def validate_contents(self, repo):
             return []
 
         content_list = github.get("/repos/" + repo["full_name"] + "/contents/")
-        if not content_list.ok:
+        # Empty repos return an object with a "message" that the repo is empty.
+        if not content_list.ok or "message" in content_list.json():
```

Comment:
I also have a fix for this about to get pushed up, but it doesn't immediately return on an empty repo. My change handles it so that the empty repo gets added to the BUNDLE_IGNORE_LIST, so that subsequent validators aren't run on the repo.

Do you want to include it with your PR, or in mine? My change:
```python
if not content_list.ok:
    if not self.validate_contents_quiet:
         return [ERROR_UNABLE_PULL_REPO_CONTENTS]

content_list = content_list.json()

#>>>> start change
files = []
# an empty repo will return a 'message'
if "message" not content_list:
    files = [x["name"] for x in content_list]
#<<<< end change

if len(files) < 8:
    BUNDLE_IGNORE_LIST.append(repo["name"])
```

Reply:
It crashed for me if I didn't return early because the list comprehension afterwards fails. If there are contents it comprehends json objects, if not, it's the key strings.

Reply:
I fudged the condition above; should be `if "message" not in content_list:`.

It worked for me. I'll go ahead and merge this, pull it into my changes and test further.